ipn/localapi/localapi.go

```diff
@@ -1200,6 +1201,24 @@ func (h *Handler) serveDial(w http.ResponseWriter, r *http.Request) {
 	<-errc
 }
 
+func (h *Handler) serveDeviceToken(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "POST" {
+		http.Error(w, "unsupported method", http.StatusMethodNotAllowed)
+		return
+	}
+	var params struct {
+		deviceToken string `json:"deviceToken"`
```

Comment:
capital DeviceToken

Reply:
just the field or also the json parameter name? we seem to have both kinds of parameter names?

Reply:
You can't json.Unmarshal (due to `reflect` rules) into a field that's not exported. JSON parameter name doesn't matter.

But really this type shouldn't be defined here like this. It should be in a package where the client side can use it, even if the sole client currently in Swift that doesn't need it.

Reply:
Yeah I changed it last minute after testing all the code.

Does the localapi have tests I should be extending for a new endpoint?

What package would the type go in?

Reply:
(the question about the parameter name was a style question, I've no opinion other than I couldn't figure out what the standard was here.)

Reply:
```go
// Package apitype contains types for the Tailscale LocalAPI and control plane API.
package apitype
```